service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/NotifyGatekeeperController.java

```diff
@@ -51,18 +52,25 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToStartEvent(@RequestBody
         return respond(caseDetails, errors);
     }
 
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmitEvent(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        if (SUBMITTED.equals(caseData.getState())) {
+            caseDetails.getData().putAll(standardDirectionsService.populateStandardDirections(caseData));
+        }
+
+        return respond(caseDetails);
+    }
+
     @PostMapping("/submitted")
     public void handleSubmittedEvent(@RequestBody CallbackRequest callbackRequest) {
         CaseData caseData = getCaseData(callbackRequest);
-        if (SUBMITTED.equals(caseData.getState())) {
-            publishEvent(new PopulateStandardDirectionsEvent(callbackRequest));
-        }
         publishEvent(new NotifyGatekeepersEvent(caseData));
     }
 
     private List<Element<EmailAddress>> resetGateKeeperEmailCollection() {
-        return List.of(
-            element(EmailAddress.builder().email("").build())
-        );
+        return wrapElements(EmailAddress.builder().email("").build());
```

Comment:
Just noticed that this doesn't add an id, might be something that needs changing. Not sure how CCD would like it.